pallets/proposals/src/impls.rs

```diff
@@ -101,6 +101,79 @@ impl<T: Config> Pallet<T> {
         Ok(().into())
     }
 
+//update new project
+pub fn update_existing_project(
+    who: T::AccountId,
+    project_key: ProjectKey,
+    name: BoundedStringField,
+    logo: BoundedStringField,
+    description: BoundedDescriptionField,
+    website: BoundedWebsiteUrlField,
+    proposed_milestones: BoundedProposedMilestones,
+    required_funds: BalanceOf<T>,
+    currency_id: common_types::CurrencyId,
+) -> DispatchResultWithPostInfo {
+    // Check if identity is required
+    if IsIdentityRequired::<T>::get() {
+        let _ = Self::ensure_identity_is_decent(&who)?;
+    }
+
+    //check to ensure valid and existing project
+    let mut project = Projects::<T>::get(&project_key).ok_or(Error::<T>::ProjectDoesNotExist)?;
+
+    ensure!(
+    project.approved_for_funding == false,
+    Error::<T>::ProjectAlreadyApproved
+    );
+
+    let mut milestone_key: u32 = 0;
+
+    let mut milestones: BTreeMap<MilestoneKey, Milestone> = BTreeMap::new();
+
+    // Fill in the projects structure in advance
+    for milestone in proposed_milestones {
+        let milestone = Milestone { 
+            project_key,
+            milestone_key,
+            name: milestone.name.to_vec(),
+            percentage_to_unlock: milestone.percentage_to_unlock,
+            is_approved: false,
+        };
+        milestones.insert(milestone_key.clone(), milestone.clone());
```

Comment:
I dont think there is any need for these clones, u32 implements the Copy trait and milestone can be owned by the BtreeMap as it is not being used again in the scope.

Reply:
Reason for this is that clones are too expensive especically on a struct and copies are quick

Reply:
Not sure I understand this. I have followed the same convention as done within Create Project. Are you suggesting revamping clone to copy across both create and upate extrinsics?